### PR TITLE
Don't require io.gatling.javaapi.core.Simulation to be on the classpath

### PR DESCRIPTION
**Pull request by nastra**
fixes https://github.com/gatling/gatling/issues/4165

### CODE REVIEW

File: src/main/java/io/gatling/mojo/GatlingMojo.java [hunk 1]
```diff
@@ -426,9 +431,10 @@ private void copyJUnitReports() throws MojoExecutionException {
         if (isIncluded && !isExcluded) {
           // check if the class is a concrete Simulation
           Class<?> clazz = testClassLoader.loadClass(className);
+          boolean isJavaSimulationClass =
+              javaSimulationClass.isPresent() && javaSimulationClass.get().isAssignableFrom(clazz);
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/GatlingMojo.java:435]: Please use `Optional#map` and `Option#orElse`.

**Review by slandelle — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Could you please address my comment and sign or [CLA](https://docs.google.com/forms/d/1hbpxVqJ5hIYYJuOjOyOzub9fJFGFGvvfnB5w5LCWdUw)?

File: src/main/java/io/gatling/mojo/GatlingMojo.java (hunk 1 above)
Comment (nastra) [reply src/main/java/io/gatling/mojo/GatlingMojo.java]: fixed

File: src/main/java/io/gatling/mojo/GatlingMojo.java [hunk 2]
```diff
@@ -427,7 +432,9 @@ private void copyJUnitReports() throws MojoExecutionException {
           // check if the class is a concrete Simulation
           Class<?> clazz = testClassLoader.loadClass(className);
           if (isConcreteClass(clazz)
-              && (javaSimulationClass.isAssignableFrom(clazz)
+              && (javaSimulationClass
+                      .map(simClass -> simClass.isAssignableFrom(clazz))
```
Comment (slandelle) [inline src/main/java/io/gatling/mojo/GatlingMojo.java:436]: wouldn't this work?

```java
.map(Class:: isAssignableFrom)
```

**Review by slandelle — CHANGES_REQUESTED** [review]

File: src/main/java/io/gatling/mojo/GatlingMojo.java (hunk 2 above)
Comment (nastra) [reply src/main/java/io/gatling/mojo/GatlingMojo.java:436]: unfortunately not as compilation fails

Comment (slandelle) [reply src/main/java/io/gatling/mojo/GatlingMojo.java:436]: sad

**Review by slandelle — APPROVED** [review]

**Comment by slandelle** [comment]
Thanks a lot!

**Pull request merged**